dotCMS/src/main/java/com/dotcms/security/secret/Secret.java

```diff
@@ -0,0 +1,44 @@
+package com.dotcms.security.secret;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public final class Secret {
+
+    private final char[] value;
+    private final boolean hidden;
+    private final SecretType secretType;
+
+    private Secret(final char[] value, final SecretType secretType, final boolean hidden) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The user-supplied array 'value' is stored directly.](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4623512)